File: tests/transpiler/x/pas/dataset_where_filter.pas
{$mode objfpc}
program Main;
uses SysUtils;
type Anon7 = record
  name: string;
  age: integer;
end;
type Anon8 = record
  name: string;
  age: integer;
  is_senior: boolean;
end;
var
  people: array of Anon7;
  adults: array of Anon8;
  person: Anon7;
begin
  people := [(name: 'Alice'; age: 30), (name: 'Bob'; age: 15), (name: 'Charlie'; age: 65), (name: 'Diana'; age: 45)];
  adults := [];
  for person in people do begin
  if person.age >= 18 then begin
  adults := concat(adults, [(name: person.name; age: person.age; is_senior: person.age >= 60)]);
end;
end;
  writeln('--- Adults ---');
  for person in adults do begin
  writeln(person.name, 'is', person.age, IfThen(person.is_senior, ' (senior)', ''));
end;
end.

File: tests/transpiler/x/pl/dataset_where_filter.pl
:- style_check(-singleton).
:- initialization(main).

main :-
    People = [{name: "Alice", age: 30}, {name: "Bob", age: 15}, {name: "Charlie", age: 65}, {name: "Diana", age: 45}],
    Adults = [{name: "Alice", age: 30, is_senior: false}, {name: "Charlie", age: 65, is_senior: true}, {name: "Diana", age: 45, is_senior: false}],
    writeln("--- Adults ---"),
    Person = {name: "Alice", age: 30, is_senior: false},
    writeln("Alice is 30 "),
    Person1 = {name: "Charlie", age: 65, is_senior: true},
    writeln("Charlie is 65  (senior)"),
    Person2 = {name: "Diana", age: 45, is_senior: false},
    writeln("Diana is 45 ").
